<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>983</width>
    <height>433</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>cvdl2019_Hw1 (P78081528)</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>361</width>
      <height>191</height>
     </rect>
    </property>
    <property name="title">
     <string>1. Calibration</string>
    </property>
    <widget class="QPushButton" name="b1_1">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>1.1 Find Corners</string>
     </property>
    </widget>
    <widget class="QPushButton" name="b1_2">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>80</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>1.2 Intrinsic</string>
     </property>
    </widget>
    <widget class="QPushButton" name="b1_4">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>130</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>1.4 Distortion</string>
     </property>
    </widget>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>30</y>
       <width>161</width>
       <height>141</height>
      </rect>
     </property>
     <property name="title">
      <string>1.3 Extrinsic</string>
     </property>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>121</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Select image</string>
      </property>
     </widget>
     <widget class="QComboBox" name="c1_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>86</width>
        <height>25</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>1</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>2</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>3</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>4</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>5</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>6</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>7</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>8</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>9</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>10</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>11</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>12</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>13</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>14</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>15</string>
       </property>
      </item>
     </widget>
     <widget class="QPushButton" name="b1_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>90</y>
        <width>141</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>1.3 Extrinsic</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>210</y>
      <width>361</width>
      <height>80</height>
     </rect>
    </property>
    <property name="title">
     <string>2. Augmented Reality</string>
    </property>
    <widget class="QPushButton" name="b2_1">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>171</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>2.1 Augmented Reality</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>30</y>
      <width>291</width>
      <height>351</height>
     </rect>
    </property>
    <property name="title">
     <string>3. Image Transformation</string>
    </property>
    <widget class="QGroupBox" name="groupBox_5">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>271</width>
       <height>311</height>
      </rect>
     </property>
     <property name="title">
      <string>3.1 Rot, scale, Translate</string>
     </property>
     <widget class="QGroupBox" name="groupBox_6">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>251</width>
        <height>211</height>
       </rect>
      </property>
      <property name="title">
       <string>Parameters</string>
      </property>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Angle:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Scale:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>100</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Tx:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>130</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Ty:</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="t3_1_angle">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>40</y>
         <width>91</width>
         <height>25</height>
        </rect>
       </property>
      </widget>
      <widget class="QLineEdit" name="t3_1_scale">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>70</y>
         <width>91</width>
         <height>25</height>
        </rect>
       </property>
      </widget>
      <widget class="QLineEdit" name="t3_1_tx">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>100</y>
         <width>91</width>
         <height>25</height>
        </rect>
       </property>
      </widget>
      <widget class="QLineEdit" name="t3_1_ty">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>130</y>
         <width>91</width>
         <height>25</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>40</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>deg</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_7">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>100</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>pixel</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_8">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>130</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>pixel</string>
       </property>
      </widget>
      <widget class="QPushButton" name="b3_1">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>160</y>
         <width>231</width>
         <height>41</height>
        </rect>
       </property>
       <property name="text">
        <string>3.1 Rotation, scaling, translation</string>
       </property>
      </widget>
     </widget>
     <widget class="QPushButton" name="b3_2">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>250</y>
        <width>231</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>3.2 Perspective Transform</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_7">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>300</y>
      <width>361</width>
      <height>80</height>
     </rect>
    </property>
    <property name="title">
     <string>4. Find Contour</string>
    </property>
    <widget class="QPushButton" name="b4_1">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>171</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>4.1 Find Contour</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_8">
    <property name="geometry">
     <rect>
      <x>690</x>
      <y>20</y>
      <width>281</width>
      <height>361</height>
     </rect>
    </property>
    <property name="title">
     <string>5. Train Cifar-10 Classifier Using LeNet-5</string>
    </property>
    <widget class="QPushButton" name="b5_1">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>261</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>5.1 Show Train Images</string>
     </property>
    </widget>
    <widget class="QPushButton" name="b5_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>261</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>5.2 Show Hyperparameters</string>
     </property>
    </widget>
    <widget class="QPushButton" name="b5_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>150</y>
       <width>261</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>5.3 Train 1 Epoch</string>
     </property>
    </widget>
    <widget class="QPushButton" name="b5_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>210</y>
       <width>261</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>5.4 Show Training Result</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>270</y>
       <width>121</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Test Image Index:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="t5_5_index">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>270</y>
       <width>141</width>
       <height>25</height>
      </rect>
     </property>
     <property name="inputMask">
      <string/>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <property name="placeholderText">
      <string>(0~9999)</string>
     </property>
    </widget>
    <widget class="QPushButton" name="b5_5">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>310</y>
       <width>261</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>5.5 Inference</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>983</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
